packages/xchain-ethereum/src/client.ts

```diff
@@ -430,40 +511,56 @@ export default class Client implements XChainClient, EthereumClient {
    */
   getTransactionData = async (txId: string, assetAddress?: Address): Promise<Tx> => {
     try {
-      let tx
-      const txInfo = await this.etherscan.getTransaction(txId)
-
-      if (txInfo) {
-        if (assetAddress) {
-          tx =
-            (
-              await etherscanAPI.getTokenTransactionHistory({
-                baseUrl: this.etherscan.baseUrl,
-                assetAddress,
-                startblock: txInfo.blockNumber,
-                endblock: txInfo.blockNumber,
-                apiKey: this.etherscan.apiKey,
-              })
-            ).filter((info) => info.hash === txId)[0] ?? null
+      if (this.getNetwork() === 'mainnet') {
+        // use ethplorerAPI for mainnet - ignore assetAddress
+        const txInfo = await ethplorerAPI.getTxInfo(this.ethplorerUrl, txId, this.ethplorerApiKey)
+
+        if (txInfo.operations && txInfo.operations.length > 0) {
+          const tx = getTxFromEthplorerTokenOperation(txInfo.operations[0])
+          if (!tx) {
+            throw new Error('Need to provide valid txId')
```

Comment:
`'Need to provide valid txId'` -> `Error('Could not parse transaction data')` or similar.